src/link.cpp

```diff
@@ -1805,7 +1821,9 @@ static void construct_linker_job_elf(LinkJob *lj) {
     }
 
     // crt end
-    if (lj->link_in_crt && target_libc_needs_crti_crtn(g->zig_target)) {
+    if (target_is_android(g->zig_target) && g->libc != nullptr) {
```

Comment:
What's the reasoning for `g->libc != nullptr`? I don't think it's necessary - even if you were running zig compiler on android you'd want `crtend_android.o` on the link line, right? `g->libc` is just if someone provided a custom libc.txt file with paths.

I think you want `g->link_in_crt` instead (see the below if statement). This is whether we want to link against libc.